core_backend/app/__init__.py

```diff
@@ -73,9 +77,37 @@ def create_app() -> FastAPI:
     metrics_app = create_metrics_app()
     app.mount("/metrics", metrics_app)
 
+    async def reset_quota() -> None:
+        """Reset api daily quota for all users in redis"""
+        redis = app.state.redis
+        async for asession in get_async_session():
+            try:
+                users = await get_all_users(asession)
+                for user in users:
+                    api_daily_quota = encode_api_limit(user.api_daily_quota)
+                    await redis.set(
+                        f"remaining-calls:{user.username}",
+                        api_daily_quota,
+                    )
+                logger.info("Successfully Synced Redis to database")
+            except Exception as e:
+                logger.info(f"Redis sync failed: {e}")
+
     @app.on_event("startup")
-    def startup_event() -> None:
+    async def startup_event() -> None:
```

Comment:
If we redeploy the backend app in the middle of the day, will the API limits reset?

Reply:
I updated the code to make sure we execute it once. By setting a redis value that expires at midnight. So  we only run the job when it expires